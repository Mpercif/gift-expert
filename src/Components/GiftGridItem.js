import React from 'react'
import PropTypes from 'prop-types'



const GiftGridItem = ({title, url}) => {

    return (
        <div className='card animate__bounceInRight' >
            <img src={url} alt={title} />
            <p className='textBox' >{title}</p>
        </div>
    )
}

GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GiftGridItem

