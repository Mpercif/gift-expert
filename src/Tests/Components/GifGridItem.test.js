import React from 'react';
import { shallow } from "enzyme"
import GiftGridItem from "../../Components/GiftGridItem"

describe('Modulo GifGridItem', () => {
    
    const objTest = {
        url: 'https://media3.giphy.com/media/21PccxrfPVOV1lhBAQ/giphy.gif?cid=4978c6e2rfe8uml2x9w8i9av35kqtkimm04wb2lz3zivj33c&rid=giphy.gif&ct=g"',
        title: 'Kachan'
    }
    const wraper = shallow(<GiftGridItem {...objTest} /> )

    test('Debe renderizar correctamente el item ', () => {
           
        expect({wraper}).toMatchSnapshot() 
    })

    test('Debe de tener un parrafo con el title ', () => {
        
        const p = wraper.find('p')
        expect(p.text().trim()).toBe(objTest.title)
    })

    test('Debe tener la imagen igual a la url y alt de los props ', () => {
        
        const img = wraper.find('img') 
        expect(img.prop('src')).toBe(objTest.url)
        expect(img.prop('alt')).toBe(objTest.title)
    })

    test('Debe de tener el clasname = "animate__bounceInRight" ', () => {
        
        const div = wraper.find('div')
        expect(div.hasClass('animate__bounceInRight')).toBe(true)

    })
    
    
    
    

})


