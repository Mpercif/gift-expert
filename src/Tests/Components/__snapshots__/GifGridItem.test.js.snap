// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modulo GifGridItem Debe renderizar correctamente el item  1`] = `
Object {
  "wraper": <div
    className="card animate__bounceInRight"
  >
    <img
      alt="Kachan"
      src="https://media3.giphy.com/media/21PccxrfPVOV1lhBAQ/giphy.gif?cid=4978c6e2rfe8uml2x9w8i9av35kqtkimm04wb2lz3zivj33c&rid=giphy.gif&ct=g\\""
    />
    <p
      className="textBox"
    >
      Kachan
    </p>
  </div>,
}
`;
